### DISCUSSION TITLE
Explicit Lorentz indices in a simple FeynCalc computation

### DISCUSSION THREAD

**Discussion by helfaham** [FeynCalc/feyncalc#199] [Q&A]
Hi,

Versions: Mathematica 12.3.1, FeynCalc 9.3.1, FeynArts 3.11

I am computing a simple tree-level qq->qq amplitude as follows (where I show the main parts of the code):
`diags = InsertFields[tops, {F[7], -F[7]} -> {F[9], - F[9]}, 
   InsertionLevel -> {Classes},
   Model -> "SMEFTatLO_FA",
   GenericModel -> "SMEFTatLO_FA"];`

`AmpFA = CreateFeynAmp[diags, Truncated -> False] /.
   {GS -> FAGS, 
    SUNT -> FASUNT, SUNF -> FASUNF, FeynAmp -> FAFeynAmp, 
    PropagatorDenominator -> FAPropagatorDenominator, 
    MetricTensor -> FAMetricTensor, 
    NonCommutative -> FANonCommutative, DiracSpinor -> FADiracSpinor, 
    DiracMatrix -> FADiracMatrix, 
    PolarizationVector -> FAPolarizationVector, MT -> SMP["m_t"]};`

`AmpFC =
  FCFAConvert[AmpFA, IncomingMomenta -> {p1, p2}, 
     OutgoingMomenta -> {p3, p4}, List -> True,
     ChangeDimension -> 4, SMP -> True, DropSumOver -> True, 
     UndoChiralSplittings -> True] // PropagatorDenominatorExplicit //
    Contract;`
I end up with an amplitude that contains explicit Dirac matrices holding Lorentz indices (indicated with **). I naively expect they should have been contracted in the process. Is there a way to fix this or should I substitute this explicit structure manually?
```
-((gs^2 Spinor[-Momentum[p2], 0, 1] . 
      **DiracGamma[LorentzIndex[Lor2]]** . 
      Spinor[Momentum[p1], 0, 1] Spinor[Momentum[p3], SMP["m_t"], 
       1] . **DiracGamma[LorentzIndex[Lor2]]** . 
      Spinor[-Momentum[p4], SMP["m_t"], 1] SUNTF[{SUNIndex[Glu5]}, 
      SUNFIndex[Col2], SUNFIndex[Col1]] SUNTF[{SUNIndex[Glu5]}, 
      SUNFIndex[Col3], SUNFIndex[Col4]])/(Pair[Momentum[p3], 
      Momentum[p3]] + 2 Pair[Momentum[p3], Momentum[p4]] + 
     Pair[Momentum[p4], Momentum[p4]]))
```
Thank you in advance for your help!

Best,
Hesham

**Comment by vsht** [comment]
Hi,

first of all, why don't you properly patch the model using `FAPatch[FAModelsDirectory -> "dir", PatchModelsOnly -> True]`?
The set of replacement rules you are applying will most likely fail at some point.

Second, dangling indices might well come from issues with the model itself, so checking it with FormCalc would be a good idea.

Cheers,
Vladyslav
